src/services/LiquidityService/LiquidityService.js

```diff
@@ -166,14 +169,25 @@ check should be done`,
       boughtOrSoldTokensPromise = Promise.resolve([])
     } else {
       // Ensure liquidity + Create concurrency lock
-      this.concurrencyCheck[lockName] = this[doEnsureLiquidityFnName]({
-        tokenA: sellToken,
-        tokenB: buyToken,
-        from,
-        minimumSellVolume,
-        buyLiquidityRules
-      })
-      boughtOrSoldTokensPromise = this.concurrencyCheck[lockName]
+      this.concurrencyCheck[lockName] = {
+        transaction: this[doEnsureLiquidityFnName]({
```

Comment:
This transaction, is a promise. Right? Maybe we could call it transactionPromise